### ISSUE TITLE
Switch menu to current task

### ISSUE THREAD

**Issue by vantu5z** [Schneegans/Fly-Pie#18]
Example structure:

**Home menu** (opens first time you open menu after system reboot)
- Internet
- Youtube
- Photo (editing)
- Video (editing)
- Programming
- Project 1
...
- Project N

You select one you want to do now.
And after you call menu next time will open **relevant menu**.
When you end with this task you should have some way to switch back to **home menu**.

As solution we need API (terminal command) to change shortcut for menu.

**Comment by Schneegans** [comment]
Hi there! I really struggle to understand what you are asking for. Could you try to explain the goal of your feature? Do you want to open different menus with the same hotkey, depending on which application has focus currently? Or do you want Fly-Pie to remember the last open submenu?

**Comment by vantu5z** [comment]
Try to explain.
I want to open different menus with the same hotkey. But I want to select menu depending on task.
For example I want work with photos, open menu and select this task. And now menu became photo oriented (apps, folders, commands, ...). When I finish, open menu and select exit. Now menu became "home".

**Comment by vantu5z** [comment]
Found that this can be done using DBus and Python:

Create Python file `~/.locan/bin/flypie-dbus` (code is not ideal, just for example):
```
#!/usr/bin/env python
# -*- coding: utf-8 -*-

from pydbus import SessionBus
from gi.repository import GLib
import os

fly_menu = '''{"icon": "stock_home", "children": [
                {"name": "Photo", "icon":"📷", "id":"photo"},
                {"name": "Video", "icon":"camera-video", "id":"video"},
                {"name": "Internet", "icon":"stock_internet", "id":"internet"},
                {"name": "projects", "icon":"👷", "children": [
                  {"name": "Project_1","icon":"💻", "id":"project_1"},
                  {"name": "Project_2", "icon":"🖥", "id":"project_2"}
                  ]}
                ]}'''

def on_signal(*args):
    if args[3] == 'OnSelect':
        selected_menu_name = args[4][1]
        os.system("echo '%s' > /tmp/current_FlyMenu" % selected_menu_name)
    loop.quit()

bus = SessionBus()

fly_menu_bus = bus.get("org.gnome.Shell",
                       "/org/gnome/shell/extensions/flypie")

show_home = False
try:
    conf_file = open('/tmp/current_FlyMenu', 'r')
    menu_name = conf_file.readline().replace("\n", "")
    if fly_menu_bus.ShowMenu(menu_name) < 0:
        show_home = True
except:
    show_home = True

if show_home:
    fly_menu_bus.ShowCustomMenu(fly_menu)

    # start monitoring signals
    loop = GLib.MainLoop()
    bus.subscribe(object="/org/gnome/shell/extensions/flypie",
                  signal_fired = on_signal)
    loop.run()
```

`fly_menu` contain **home-menu** description where `id` are existing FlyPie menus (**task-menu**) created by editor.
At each FlyPie **task-menu** we should add element with command "rm /tmp/current_FlyMenu" to remove the file which remember **task-menu** name.
Set hotkey to run `flypie-dbus` to open relevant menu.

I found solution, but it will be nice to have this functionality using only configuration dialog (without programming).

**Comment by Schneegans** [comment]
Great solution! I will think about a feature for opening pies depending on the application which currently has focus. I think this could solve your issue as well and might be a useful feature for others as well. Thank you for the idea!
